packages/data-context/src/codegen/spec-options.ts

```diff
@@ -54,19 +55,14 @@ export class SpecOptions {
       throw new Error('Cannot generate a spec without a framework')
     }
 
-    // This only works for Vue projects right now. If the framework is not Vue, we're generating an empty component test
-    if (this.options.framework.codeGenFramework !== 'vue') {
-      return {
-        codeGenType: this.options.codeGenType,
-        fileName: await this.buildFileName(),
-        templateKey: 'componentEmpty' as TemplateKey,
-        overrideCodeGenDir: '',
-      }
+    switch (this.options.framework.codeGenFramework) {
+      case 'react':
+        return await this.getReactSpecOptions()
+      case 'vue':
+        return await this.getVueSpecOptions()
+      default:
+        throw new Error(`Unable to generate spec for ${this.options.framework.codeGenFramework}`)
```

Comment:
Since this now errors rather than falling back to an empty spec do we need the `componentEmpty` entry in `TemplateKey` above?

Reply:
It shouldn't be possible to reach this line since the UI will only show the Create from Component generator card for React and Vue apps.

`componentEmpty` is still a valid template key because that generator card will still appear in the UI for all applications